src/hooks/useSignIn.ts

```diff
@@ -0,0 +1,48 @@
+import { GITHUB_OAUTH_CLIENT_ID } from '@env';
+import { useCallback, useState } from 'react';
+import { authorize } from 'react-native-app-auth';
+import login from '../utils/api/login';
+import { setSecureValue } from '../utils/setSecureValue';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+
+async function authorizeUser(config: Config) {
+  try {
+    const result = await authorize(config);
+    return result;
+  } catch (err) {
+    console.log(err);
+  }
+}
+
+export default function useSignIn() {
```

Comment:
Core logic for sign in with Github
1. authorization: Where a browser [Custom Tab/ SafariSafeView] opens and asks the user to sign-in to Github and then authorize the registered oAuth2 App on Github to be authorized
2. login calls the backend where the code exchange happens and client gets the accessToken, and user data
3. Access token is stored in react native keychain's secure storage, and user info in AsyncStorage